From 322b27a17760508c959a30f2ca0e764794cb50cf Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Wed, 14 Nov 2018 19:57:03 -0700
Subject: [PATCH 06/15] msm8974-common: sepolicy: Create variety of sysfs_sec_*
 types

* Rename sysfs_sec type to sysfs_sec_hall
* Add additional sysfs_sec_* types as appropriate

Change-Id: I66b6d2aab875a2706f2730be9755e8d9805ffb6e
---
 sepolicy/common/file.te       | 11 ++++++++++-
 sepolicy/common/file_contexts | 13 ++++++++++++-
 sepolicy/common/rild.te       |  2 +-
 3 files changed, 23 insertions(+), 3 deletions(-)

diff --git a/sepolicy/common/file.te b/sepolicy/common/file.te
index 87ef68a..52a6e4b 100644
--- a/sepolicy/common/file.te
+++ b/sepolicy/common/file.te
@@ -5,7 +5,16 @@ type sysfs_hal_pwr, fs_type, sysfs_type;
 type sysfs_iio, fs_type, sysfs_type;
 type sysfs_input, fs_type, sysfs_type;
 type sysfs_mdnie, fs_type, sysfs_type;
-type sysfs_sec, fs_type, sysfs_type;
+type sysfs_sec_bamdmux, fs_type, sysfs_type;
+type sysfs_sec_barcode_emul, fs_type, sysfs_type;
+type sysfs_sec_epen, fs_type, sysfs_type;
+type sysfs_sec_ir, fs_type, sysfs_type;
+type sysfs_sec_key, fs_type, sysfs_type;
+type sysfs_sec_led, fs_type, sysfs_type;
+type sysfs_sec_switch, fs_type, sysfs_type;
+type sysfs_sec_thermistor, fs_type, sysfs_type;
+type sysfs_sec_touchkey, fs_type, sysfs_type;
+type sysfs_sec_tsp, fs_type, sysfs_type;
 type sysfs_wifi_writeable, fs_type, sysfs_type;
 
 type bt_fw_file, file_type;
diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index 3415576..06d65fc 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -33,7 +33,6 @@
 /sys/devices/platform/bcm[0-9]+_bluetooth/rfkill/rfkill0/state   u:object_r:sysfs_bluetooth_writable:s0
 /sys/devices/virtual/camera(/.*)?                                u:object_r:sysfs_camera:s0
 /sys/devices/virtual/input(/.*)?                                 u:object_r:sysfs_input:s0
-/sys/devices/virtual/sec/sec_key/hall_irq_ctrl                   u:object_r:sysfs_sec:s0
 /sys/module/dhd/parameters/firmware_path                         u:object_r:sysfs_wifi_writeable:s0
 /sys/module/dhd/parameters/nvram_path                            u:object_r:sysfs_wifi_writeable:s0
 
@@ -60,3 +59,15 @@
 
 # sysfs - mdnie
 /sys/devices/virtual/mdnie/mdnie(/.*)?                  u:object_r:sysfs_mdnie:s0
+
+# sysfs - sec
+/sys/devices/platform/sec-thermistor(/.*)?              u:object_r:sysfs_sec_thermistor:s0
+/sys/devices/virtual/sec/bamdmux(/.*)?                  u:object_r:sysfs_sec_bamdmux:s0
+/sys/devices/virtual/sec/led(/.*)?                      u:object_r:sysfs_sec_led:s0
+/sys/devices/virtual/sec/sec_barcode_emul(/.*)?         u:object_r:sysfs_sec_barcode_emul:s0
+/sys/devices/virtual/sec/sec_epen(/.*)?                 u:object_r:sysfs_sec_epen:s0
+/sys/devices/virtual/sec/sec_ir(/.*)?                   u:object_r:sysfs_sec_ir:s0
+/sys/devices/virtual/sec/sec_key(/.*)?                  u:object_r:sysfs_sec_key:s0
+/sys/devices/virtual/sec/sec_touchkey(/.*)?             u:object_r:sysfs_sec_touchkey:s0
+/sys/devices/virtual/sec/switch(/.*)?                   u:object_r:sysfs_sec_switch:s0
+/sys/devices/virtual/sec/tsp(/.*)?                      u:object_r:sysfs_sec_tsp:s0
diff --git a/sepolicy/common/rild.te b/sepolicy/common/rild.te
index 6bbe2cf..8efef11 100644
--- a/sepolicy/common/rild.te
+++ b/sepolicy/common/rild.te
@@ -5,4 +5,4 @@ allow rild radio_data_file:file create_file_perms;
 allow rild radio_data_file:lnk_file read;
 
 allow rild proc_net:file w_file_perms;
-allow rild sysfs_sec:file rw_file_perms;
+allow rild sysfs_sec_key:file rw_file_perms;
-- 
2.17.1

